src/sdf.cpp

```diff
@@ -525,7 +525,9 @@ Manifold Manifold::LevelSet(std::function<double(vec3)> sdf, Box bounds,
   triVerts.resize(index[0]);
 
   pImpl_->CreateHalfedges(triVerts);
+  pImpl_->RemoveUnreferencedVerts();
   pImpl_->CleanupTopology();
+  pImpl_->RemoveUnreferencedVerts();
```

Comment:
This seems wrong - having both of these was the only way I could fix the seg fault in `BooleanComplex.MeshRelation`, but it seems it is still happening on the CI anyway. I haven't been able to figure out what's going wrong here because even with address sanitizer I can't make it fail when run alone - only when the whole test suite is run, which doesn't really work for debug. 

Any thoughts @pca006132? I'm beginning to think we should get a bit more crisp about where unreferenced verts can get created and where they get removed. But maybe there's some other problem...

Reply:
No idea why it only fails when we run the whole test suite. I think we should probably have some validation code that track this unreferenced vertex information. Should be more reliable than asan.

Reply:
Turns out we ran into exactly this problem before, in #894. I forgot that address sanitizer doesn't check for uninitialized memory.